register-machines/src/main.rs

```diff
@@ -13,66 +15,165 @@ enum Instruction {
 
 use Instruction::*;
 fn eval_program(program: &[Instruction], state: &State) -> State {
-    unimplemented!();
+    let (mut l, mut rs) = state.clone();
+    while l < program.len() {
+        match program[l] {
+            Add(i, j) => {
+                *rs.entry(i).or_insert(Godel::new()) += 1;
+                l = j; 
+            },
+            Sub(i, j, k) => {
+                let v = rs.entry(i).or_insert(Godel::new()); 
```

Comment:
`let v = rs.entry(i).or_insert_with(Godel::new);` is typically seen as more idiomatic. Probably something to do with optimisation. It is what it is I guess.
Yet another stylistic change:
`let v = rs.entry(i).and_modify(|e| *e - 1).or_insert_with(Godel::new);`